CenterForOpenScience/ember-osf-web
# HTML for File Detail Page

-   Ticket: https://openscience.atlassian.net/browse/ENG-3517
-   Feature flag: n/a

## Purpose

The purpose of this change was to create the initial html from the UX Pin Mock ups shared with us by Product and Design. 

## Summary of Changes

The guid file route has been updated for the detail view, to include left and right panels with a side nav bar. This navbar has two buttons: one to expand version data and tags button which is still awaiting final mock-ups from product. The version data displays a version header, version number currently as a template string and date created. The dropdown content for each version expands to allow the user to copy the MD5 and SHA-2 checksums and download a copy of the file. The tooltips will be updated to display user-friendly strings. 

## Screenshot(s)

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/82047646/155366571-2e3077bf-6794-4230-a26f-0cb8be6a2588.png">
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/82047646/155366620-7ccf70ac-063f-4c41-a786-9dc1b8bf0308.png">
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/82047646/155366696-fa0457ef-89a3-4492-84d4-53928a4383d9.png">
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/82047646/155366731-3a7d5e52-9c63-441f-a0c7-b2995dd62fc3.png">
Currently in debug:
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/82047646/155366872-50dc84df-043e-4f50-bfae-2c4170a651fd.png">

Note: Screenshots being updated with progress on CSS.

## Side Effects

The layout of the page has been fully updated, and should be tested for dropdown elements to include the download and share buttons; that the project link redirects properly, the file is rendered by the file renderer and that the navbar properly displays on both monitor, tablet, and mobile displays. Further, the English version of the template strings, copy checksum buttons and tables should be checked for 1) existence and 2) proper UI display and 3) content.

## QA Notes

    - Does the project link properly redirect the user?
    - Verify file name and version displays
    - Does the version link work correctly?
    - Does the dropdown button work? Does it display buttons vertically?
    - Does the file render? 
    - Are the buttons functional? Is there any logic that needs testing (versions, permissions, etc)
    - Are versions present? If so, do the links properly redirect the user?
    -  Can you download a functional version of the file?
    - Can you verify its checksums?
    - Do the tooltips properly display helpful strings? Are they too long? Should and wording be rephrased?
